scripts/up.sh

```diff
@@ -63,17 +23,58 @@ else
   exit 1
 fi
 
-echo "Bringing up network with '$TYPE' configuration."
+echo "Before bootrapping, try cleaning up first..."
 
-# Make sure it isn't running
-./scripts/down.sh
+# Make sure it isn't running and clean up any docker detritous
+./scripts/down.sh -f "$ENV_FILE"
+rm -rf "$SANDBOX_DIR"
+echo "up.sh. seconds it took to get to end of $SANDBOX_DIR cleanup: " + $(($(date "+%s") - $START))
 
-# Remove the containers to allow re-running stateful tests.
-docker-compose rm --force
+SANDBOX_CFG="_config.harness"
+cp config.harness "$SANDBOX_CFG"
+if [[ $TYPE == "channel" ]]; then
+  ALGOD_URL=""
+  ALGOD_BRANCH=""
+  ALGOD_SHA=""
+else
+  ALGOD_CHANNEL=""
+fi
 
-# When developing, it's often useful to skip the build phase.
-if [[ $SKIP_BUILD -eq 0 ]]; then
-  docker-compose build --no-cache $PARALLEL
+if [[ $OSTYPE == 'darwin'* ]]; then
+  SED_CMD='sed -i "" '
+else
+  SED_CMD='sed -i '
 fi
 
-docker-compose up $DAEMON_FLAG
+echo "On this OSTYPE=$OSTYPE using sed variant $SED_CMD"
+
+echo "apply regex replacements on SANDBOX_CFG=$SANDBOX_CFG"
+$SED_CMD "s|#ALGOD_CHANNEL|$ALGOD_CHANNEL|g" "$SANDBOX_CFG"
+$SED_CMD "s|#ALGOD_URL|$ALGOD_URL|g" "$SANDBOX_CFG"
+$SED_CMD "s|#ALGOD_BRANCH|$ALGOD_BRANCH|g" "$SANDBOX_CFG"
+$SED_CMD "s|#ALGOD_SHA|$ALGOD_SHA|g" "$SANDBOX_CFG"
+$SED_CMD "s|#NETWORK_TEMPLATE|$NETWORK_TEMPLATE|g" "$SANDBOX_CFG"
+$SED_CMD "s|#NETWORK_NUM_ROUNDS|$NETWORK_NUM_ROUNDS|g" "$SANDBOX_CFG"
+$SED_CMD "s|#NODE_ARCHIVAL|$NODE_ARCHIVAL|g" "$SANDBOX_CFG"
+$SED_CMD "s|#NODE_V1_INDEXER|$NODE_V1_INDEXER|g" "$SANDBOX_CFG"
+$SED_CMD "s|#INDEXER_URL|$INDEXER_URL|g" "$SANDBOX_CFG"
+$SED_CMD "s|#INDEXER_BRANCH|$INDEXER_BRANCH|g" "$SANDBOX_CFG"
+$SED_CMD "s|#INDEXER_SHA|$INDEXER_SHA|g" "$SANDBOX_CFG"
+
+echo "Resulting $SANDBOX_CFG:"
+cat "$SANDBOX_CFG"
+
+rootdir=$(dirname "$0")
+pushd "$rootdir"/.. > /dev/null || exit
+
+git clone --branch "$SANDBOX_BRANCH" --single-branch "$SANDBOX_URL" "$SANDBOX_DIR"
+
+echo "up.sh. seconds it took to get to end of cloning sandbox into $SANDBOX_DIR: " + $(($(date "+%s") - $START))
+
+echo "Bringing up network with '$TYPE' configuration."
+cp .env "$SANDBOX_DIR"/.
```

Comment:
TODO: do I still need this `.env`? Seems like I'm duplicating configuration and that ought to be able to just stick with `config.harness` generation and remove the dummy `.env` just added to sandbox.

Reply:
I got convinced that I need it, mainly because [this conversation](https://github.com/algorand/py-algorand-sdk/pull/367#discussion_r941331468).

TLDR - this repo should be able to standup a test-harness with a simple call to `up.sh`. This might even become a github-actions based integrations test.

Reply:
@michaeldiamant